node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeadmin/NodeAdminImpl.java

```diff
@@ -105,8 +134,8 @@ public boolean setFrozen(boolean wantFrozen) {
         }
 
         // Use filter with count instead of allMatch() because allMatch() will short circuit on first non-match
-        boolean allNodeAgentsConverged = nodeAgentsByHostname.values().stream()
-                .filter(nodeAgent -> !nodeAgent.setFrozen(wantFrozen))
+        boolean allNodeAgentsConverged = nodeAgentWithSchedulerByHostname.values().parallelStream()
+                .filter(nodeAgentScheduler -> !nodeAgentScheduler.setFrozen(wantFrozen, NODE_AGENT_FREEZE_TIMEOUT))
```

Comment:
Right, I changed it to `parallelStream()` for that reason. Given that the method is lightweight and most of the time will be spent sleeping, I assume that it would be able to start the wait for all of them at (nearly) the same time?

Reply:
I think so. So the parallelstream will spawn up to 32 threads to accomplish this? Isn't it implementation defined and it may spawn less, etc? I mean, there had been millions of elements it wouldn't spawed that many threads - there's got to be some limitation. Anyway, parallelStream() is probably fine.

Reply:
A thread put to sleep in cannot start executing the next setFrozen. (Well, they can with 'fibers', which I presume Java doesn't implement yet (project Loom))